src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/TagHelperCompletionProvider.cs

```diff
@@ -54,20 +54,20 @@ internal class TagHelperCompletionProvider : RazorCompletionItemProvider
         }
 
 
-        public override IReadOnlyList<RazorCompletionItem> GetCompletionItems(RazorSyntaxTree syntaxTree, TagHelperDocumentContext tagHelperDocumentContext, SourceSpan location)
+        public override IReadOnlyList<RazorCompletionItem> GetCompletionItems(RazorCompletionContext context, SourceSpan location)
         {
-            if (syntaxTree is null)
+            if (context is null)
             {
-                throw new ArgumentNullException(nameof(syntaxTree));
+                throw new ArgumentNullException(nameof(context));
             }
 
-            if (tagHelperDocumentContext is null)
+            if (context.TagHelperDocumentContext is null)
```

Comment:
Interesting, do we ever actually pass `null`? Wondering if we should make it non-nullable on the context.

Wont leave comments on the other completion providers in this PR but depending on the answer it'd be nice to remove the check 😄

Reply:
I had left it because there were a ton of tests that passed a null TagHelperDocumentContext, but it turns out that's more of a historical curiosity than a hard requirement, and I can make a dumb-y TagHelperDocumentContext fairly easily.